nrfx_log.h

```diff
@@ -36,6 +36,35 @@
 extern "C" {
 #endif
 
+#define NRFX_MODULE_PREFIX  _CONCAT(NRFX_, NRFX_LOG_MODULE)
+/*
+ * The following macros from nrfx_config control the log messages coming from
+ * a given module:
+ * - NRFX_<module>_CONFIG_LOG_ENABLED enables the messages (when set to 1)
+ * - NRFX_<module>_CONFIG_LOG_LEVEL specifies the severity level of the messages
+ *   that are to be output.
+ */
+#if !IS_ENABLED(_CONCAT(NRFX_MODULE_PREFIX, _CONFIG_LOG_ENABLED))
+#define NRFX_MODULE_CONFIG_LOG_LEVEL 0
+#else
+#define NRFX_MODULE_CONFIG_LOG_LEVEL \
+	_CONCAT(NRFX_MODULE_PREFIX, _CONFIG_LOG_LEVEL)
+#endif
+
+#if	NRFX_MODULE_CONFIG_LOG_LEVEL == 0
+#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_NONE
+#elif	NRFX_MODULE_CONFIG_LOG_LEVEL == 1
+#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_ERR
+#elif	NRFX_MODULE_CONFIG_LOG_LEVEL == 2
+#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_WRN
+#elif	NRFX_MODULE_CONFIG_LOG_LEVEL == 3
+#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_INF
+#elif	NRFX_MODULE_CONFIG_LOG_LEVEL == 4
+#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_DBG
+#endif
```

Comment:
What about:
```
#if     defined(NRFX_MODULE_CONFIG_LOG_LEVEL_ERR)
#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_ERR
#elif	defined(NRFX_MODULE_CONFIG_LOG_LEVEL_WRN)
#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_WRN
#elif	defined(NRFX_MODULE_CONFIG_LOG_LEVEL_INF)
#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_INF
#elif	defined(NRFX_MODULE_CONFIG_LOG_LEVEL_DBG)
#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_DBG
#else
#define NRFX_MODULE_LOG_LEVEL		LOG_LEVEL_NONE
#endif

Reply:
But how are these `NRFX_MODULE_CONFIG_LOG_LEVEL_*` symbols supposed to be defined?

Reply:
@emob-nordic I see no way of generic checking if a label `NRFX_*_CONFIG_LOG_LEVEL_*` is defined for a given module, so that you could specify, for instance:
```
zephyr_compile_definitions(NRFX_I2S_CONFIG_LOG_ENABLED=1)
zephyr_compile_definitions(NRFX_I2S_CONFIG_LOG_LEVEL_DBG)
```
to enable debug messages for the nrfx_i2s driver.

What I think would be achievable (haven't tested it yet) is to use values for these `_LOG_LEVEL_*` labels, so you would need to use:
```
zephyr_compile_definitions(NRFX_I2S_CONFIG_LOG_ENABLED=1)
zephyr_compile_definitions(NRFX_I2S_CONFIG_LOG_LEVEL_DBG=1)
```
But I'm not convinced if this would be convenient. I'm afraid that it could be rather confusing.

Reply:
Yes, you are right in this matter. Let's have this done as You proposed which is much cleaner.